android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FileUploadsDirCleaner.kt

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.file
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.support.base.log.logger.Logger
+import java.io.File
+import java.io.IOException
+
+/**
+ * A storage implementation for organizing temporal uploads metadata.
+ */
+object FileUploadsDirCleaner {
+
+    @VisibleForTesting
+    internal var fileNamesToBeDeleted: List<String> = emptyList()
+    private val scope = CoroutineScope(Dispatchers.IO)
+    val logger = Logger("UploadMetadataStorage")
+
+    /**
+     * Enqueue the [fileName] for future clean up.
+     */
+    fun enqueueForCleanup(fileName: String) {
```

Comment:
This and `cleanRecentUploads` seem to be only use in this module so maybe they can be internal